File: 890-lemonade-change/lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int five=0,ten=0,twenty=0;
        for(auto it:bills){

            if(it==5)
            {five++;}
            else if(it==10){
                ten++;
                if(five>0)
                {
                    five--;
                }
                else return false;
            }
            else if(it==20){
                twenty++;
                if(ten>0&&five>0)
                {
                    ten--;
                    five--;
                }
                else if(five>2){
                    five-=3;
                }
                else
                return false;
            }
        }
        return true;
        
    }
};

File: 890-lemonade-change/lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int five=0;
        int ten=0;
        int twenty=0;
        for(int i=0;i<bills.length;i++){
            if(bills[i]==5)five++;
            if(bills[i]==10){
                
                if(five<=0)return false;
                ten++;
                five--;
                }
            if(bills[i]==20){
            if(five<=0&&ten<=0)return false;
            if(five>=1&&ten>=1){
                five--;
                ten--;
                twenty++;
            }
            else if(five>=3){
                five--;
                five--;
                five--;
                twenty++;
            }else return false;
            }

        
        }
        return true;
    }
}